TFEngine.py

```diff
@@ -127,7 +122,17 @@ def callback_on_new():
         self.engine._checked_uninitialized_vars = False
         self.engine.updater.init_optimizer_vars(session=self.engine.tf_session)
 
-      d["optim_op"] = self.engine.updater.get_optim_op(callback_on_new=callback_on_new)
+      optim_op = self.engine.updater.get_optim_op(callback_on_new=callback_on_new)
```

Comment:
Can you add a comment and explain why this should be called before `get_fetches_dict`?

Reply:
Done.